Gifski/SavePanelAccessoryViewController.swift

```diff
@@ -145,12 +146,51 @@ final class SavePanelAccessoryViewController: NSViewController {
 			self.estimateFileSize()
 		}
 
-		widthTextField.delegate = self
-		heightTextField.delegate = self
+		frameRateSlider.maxValue = videoMetadata.frameRate.clamped(to: 5...30)
+		frameRateSlider.doubleValue = defaultFrameRate(inputFrameRate: videoMetadata.frameRate)
+		frameRateSlider.triggerAction()
 
-		// Set initial defaults
-		configureFramerateSlider(inputFrameRate: videoMetadata.frameRate)
-		configureQualitySlider()
+		qualitySlider.doubleValue = defaults[.outputQuality]
+		qualitySlider.triggerAction()
+	}
+
+	private func setupWidthAndHeightTextFields() {
+		widthTextField.onBlur = { [weak self] width in
+			self?.resizableDimensions.resize(usingWidth: CGFloat(width))
+			self?.dimensionsUpdated()
+			self?.predefinedSizesDropdown.selectItem(at: 0)
+		}
+		widthTextField.onTextDidChange = { [weak self] width in
+			guard let self = self else {
+				return
+			}
+
+			let validated = self.resizableDimensions.validate(newWidth: CGFloat(width)) == true
+			if validated {
+				self.resizableDimensions.resize(usingWidth: CGFloat(width))
+				self.dimensionsUpdated()
+				self.predefinedSizesDropdown.selectItem(at: 0)
+			} else {
+				self.widthTextField.shake(direction: .horizontal)
+			}
+		}
+		heightTextField.onBlur = { [weak self] height in
+			self?.resizableDimensions.resize(usingHeight: CGFloat(height))
+		}
+		heightTextField.onTextDidChange = { [weak self] height in
+			guard let self = self else {
+				return
+			}
+
+			let validated = self.resizableDimensions.validate(newHeight: CGFloat(height)) == true
+			if validated {
+				self.resizableDimensions.resize(usingHeight: CGFloat(height))
+				self.dimensionsUpdated()
+				self.predefinedSizesDropdown.selectItem(at: 0)
+			} else {
+				self.heightTextField.shake(direction: .horizontal)
```

Comment:
I think this should be a method on the `IntTextField`. Like `.indicateValidationFailure()` or something better.